File: backend-for-all/backend/src/main/java/com/nexaerp/banking/services/BankAccountService.java
package com.nexaerp.banking.services;

import com.nexaerp.banking.dto.BankAccountRequestDto;
import com.nexaerp.banking.dto.BankAccountResponseDto;
import com.nexaerp.banking.enums.BankAccountType;

import java.util.List;

public interface BankAccountService {
    BankAccountResponseDto create(BankAccountRequestDto request);
    BankAccountResponseDto update(Long id, BankAccountRequestDto request);
    BankAccountResponseDto getById(Long id);
    List<BankAccountResponseDto> getAll();
    List<BankAccountResponseDto> getByType(BankAccountType type);
    void deactivate(Long id);
    void activate(Long id);
}

File: accounting-system-web-project/nexa-erp-frontend/src/app/features/banking/services/bank-account.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { APP_CONFIG } from '../../../core/config/app.config';
import { ApiResponse } from '../../../core/models/api-response.model';
import { BankAccount, BankAccountRequest, BankAccountType } from '../models/bank-account.model';

@Injectable({
  providedIn: 'root',
})
export class BankAccountService {
  private readonly baseUrl = `${APP_CONFIG.apiUrl}/bank-accounts`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<ApiResponse<BankAccount[]>> {
    return this.http.get<ApiResponse<BankAccount[]>>(this.baseUrl);
  }

  getById(id: number): Observable<ApiResponse<BankAccount>> {
    return this.http.get<ApiResponse<BankAccount>>(`${this.baseUrl}/${id}`);
  }

  getByType(type: BankAccountType): Observable<ApiResponse<BankAccount[]>> {
    return this.http.get<ApiResponse<BankAccount[]>>(`${this.baseUrl}/type/${type}`);
  }

  create(request: BankAccountRequest): Observable<ApiResponse<BankAccount>> {
    return this.http.post<ApiResponse<BankAccount>>(this.baseUrl, request);
  }

  update(id: number, request: BankAccountRequest): Observable<ApiResponse<BankAccount>> {
    return this.http.put<ApiResponse<BankAccount>>(`${this.baseUrl}/${id}`, request);
  }

  deactivate(id: number): Observable<ApiResponse<null>> {
    return this.http.patch<ApiResponse<null>>(`${this.baseUrl}/${id}/deactivate`, {});
  }

  activate(id: number): Observable<ApiResponse<null>> {
    return this.http.patch<ApiResponse<null>>(`${this.baseUrl}/${id}/activate`, {});
  }
}
